bump client_golang to v1.6.0

RainbowMango opened the pull request:
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
We need new functionality that is in a later version of [prometheus/client_golang@v1.16.0](https://github.com/prometheus/client_golang/releases/tag/v1.6.0).

> [FEATURE] testutil: Add lint checks for metrics, including a sub-package promlint to expose the linter engine for external usage.

We need the `testutil` to check metrics in k/k.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

RainbowMango commented:
The unit test panic happened here:
https://github.com/kubernetes/kubernetes/blob/7229e13036e4c89c5b4977ec968229a214ce2d0c/staging/src/k8s.io/client-go/discovery/discovery_client_test.go#L474

The method `XXX_Size()` implementation of `openapi_v2.Document` depends on `github.com/golang/protobuf`'s struct `type InternalMessageInfo struct`. But after update `github.com/golang/protobuf` from v1.3.3 to v1.4.0, the `type InternalMessageInfo struct` has been deprecated.

@liggitt 
Cloud you please take an eye on this?
I saw you updated `github.com/googleapis/gnostic` to v0.4.1 in #89704 recently, should we update it again to a newer pseudo version?

<details>
  <summary>unit test panic back trace</summary>
2020/04/29 04:02:25 http: panic serving 127.0.0.1:52272: not implemented
goroutine 455 [running]:
net/http.(*conn).serve.func1(0xc0000ede00)
	GOROOT/src/net/http/server.go:1767 +0x147
panic(0x1a1f160, 0x1dc6440)
	GOROOT/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/vendor/github.com/golang/protobuf/proto.(*InternalMessageInfo).Size(...)
	vendor/github.com/golang/protobuf/proto/deprecated.go:91
k8s.io/kubernetes/vendor/github.com/googleapis/gnostic/openapiv2.(*Document).XXX_Size(0xc0001f57a0, 0x1dfed40)
	vendor/github.com/googleapis/gnostic/openapiv2/OpenAPIv2.pb.go:551 +0x48
k8s.io/kubernetes/vendor/github.com/gogo/protobuf/proto.Marshal(0x1dfed40, 0xc0001f57a0, 0x1c211ea, 0x44, 0x1ddba80, 0xc000418690, 0x52f026)
	vendor/github.com/gogo/protobuf/proto/table_marshal.go:2938 +0x3cb
k8s.io/kubernetes/vendor/k8s.io/client-go/discovery.openapiSchemaDeprecatedFakeServer.func1(0x1e0b4c0, 0xc0001c6e00, 0xc0000e8500)
	staging/src/k8s.io/client-go/discovery/discovery_client_test.go:474 +0x10c1
net/http.HandlerFunc.ServeHTTP(0xc0003dcae0, 0x1e0b4c0, 0xc0001c6e00, 0xc0000e8500)
	GOROOT/src/net/http/server.go:2007 +0x52
net/http.serverHandler.ServeHTTP(0xc0001c6540, 0x1e0b4c0, 0xc0001c6e00, 0xc0000e8500)
	GOROOT/src/net/http/server.go:2802 +0xcf
net/http.(*conn).serve(0xc0000ede00, 0x1e106c0, 0xc00042d840)
	GOROOT/src/net/http/server.go:1890 +0x838
created by net/http.(*Server).Serve
	GOROOT/src/net/http/server.go:2928 +0x5b6
</details>

RainbowMango commented:
By the way, this PR just intended to update `prometheus/client_golang` to @v1.6.0, the others are all the transitive dependencies.

RainbowMango commented:
@dims 
Seems [github.com/gogo/protobuf](https://github.com/gogo/protobuf/issues/676) not active anymore, should we remove it out from k/k?

dims commented:
@RainbowMango Jordan may be a better bet to ask this question, looks like it is all over the place in generated code

apelisse commented:
I don't think gogo/protobuf is dead, and certainly don't hope so since we're extensively using it. I'm not convinced that the alternative (golang/protobuf) is good enough for now, but we should certainly consider it at some point, though that's a significant effort.

Let me look at your change.

apelisse commented:
> I saw you updated github.com/googleapis/gnostic to v0.4.1 in #89704 recently, should we update it again to a newer pseudo version?

v0.4.1 is the last tagged release as far as I can tell. It looks like they fixed that code in master though.

apelisse commented:
https://github.com/golang/protobuf/blob/e9dc0d72c1a35f349f6d6871842199ddaaaf9edc/proto/deprecated.go#L85-L92

I don't think that's what deprecated means ;-)

apelisse commented:
@RainbowMango Do you happen to know what's forcing you to update the `golang/protobuf` dependency to 1.4?

RainbowMango commented:
> @RainbowMango Do you happen to know what's forcing you to update the `golang/protobuf` dependency to 1.4?

@apelisse That is `prometheus/client_golang`, [go.mod](https://github.com/prometheus/client_golang/blob/6edbbd9e560190e318cdc5b4d3e630b442858380/go.mod#L6),  using `github.com/golang/protobuf v1.4.0`.

RainbowMango commented:
Pushed a new commit, just replace gogo/protocol with golang/protocol when marshal a openapi document. 

https://github.com/kubernetes/kubernetes/blob/863ce9726e0aa617f03332f42ff3cebb7abe8fc0/staging/src/k8s.io/client-go/discovery/discovery_client_test.go#L474

https://github.com/kubernetes/kubernetes/blob/863ce9726e0aa617f03332f42ff3cebb7abe8fc0/staging/src/k8s.io/client-go/discovery/discovery_client_test.go#L501

RainbowMango commented:
/test pull-kubernetes-e2e-kind-ipv6
/test pull-kubernetes-verify
/test pull-kubernetes-cross

RainbowMango commented:
re-run fake testing.
/test pull-kubernetes-cross

/priority backlog

RainbowMango commented:
/assign @liggitt @brancz
Cloud you please take an eye on it? This is part of #86477.

cc guys who may be interested.
@beorn7 @logicalhan

brancz commented:
Looks good regarding the prometheus changes. Needs another pair of eyes for the rest I would say.

/lgtm

beorn7 commented:
Nit: prometheus/client_golang is v1.6.0, not v1.16.0 as in the PR title.

The Prometheus changes LGTM. I'm looking forward how you'll resolve the protobuf issue. :-/

RainbowMango commented on staging/src/k8s.io/client-go/discovery/discovery_client_test.go:
```diff
@@ -26,7 +26,7 @@ import (
 	"time"
 
-	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/proto"
```
Just replace gogo/protobuf with golang/protobuf here. 
@beorn7

liggitt replied:
I'm a bit confused by this… is this an example of switching all usage, or was this specific location causing a problem?

RainbowMango replied:
> is this an example of switching all usage

Not really. 
The `gogo/protobuf/proto.Marshal()` can not marshal the  `openapi_v2.Document` anymore if we update `github.com/golang/protobuf` to v1.4.0, because `openapi_v2.Document` depends on `github.com/golang/protobuf`'s lower version.

I've tried update `github.com/googleapis/gnostic` to a newer pseudo version(on master), but it doesn't work, I didn't dig on it.

RainbowMango replied:
One more thing, I noticed the openapi_v2.Document is Unmarshalled by the `github.com/golang/protobuf/proto` package:
https://github.com/kubernetes/kubernetes/blob/9380cb1b137b05ff12cf5a2f733653be82ff7657/staging/src/k8s.io/client-go/discovery/discovery_client.go#L437

I think it'd be better to Marshal the openapi_v2.Document  by the same package.

liggitt replied:
> The gogo/protobuf/proto.Marshal() can not marshal the openapi_v2.Document anymore if we update github.com/golang/protobuf to v1.4.0, because openapi_v2.Document depends on github.com/golang/protobuf's lower version.

Do you have more details about this? This PR doesn't modify github.com/gogo/protobuf/proto or github.com/googleapis/gnostic/openapiv2 ... what changed that caused github.com/gogo/protobuf to break? we use github.com/gogo/protobuf in many other places, are any of them affected?

liggitt replied:
According to the 1.4.0 release notes:

> This release maintains backwards compatibility with previous releases of this module. Any observable changes in behavior are to fix bugs, change unspecified behavior, or to make behavior more compliant with the protobuf specification.

So the breakage you describe sounds unexpected

RainbowMango replied:
There are 3 commits in this PR.
The first commit focus on update client_golang, and the second commit focus on update transitive dependencies.

Then I met a panic as I mentioned [before](https://github.com/kubernetes/kubernetes/pull/90582#issuecomment-621172241) where I also put a folded `unit test panic back trace` too.

The panic message is `not implemented`, which dumped from golang/protobuf@v1.4.0:
https://github.com/golang/protobuf/blob/1b794fe86dd6a0c7c52ae69b5c9cb0aeedc52afa/proto/deprecated.go#L91

Even though the golang/protobuf@v1.4.0 maintains backward compatibility with previous releases, I guess that means all exported things are still keep exported, but the implementation of [type InternalMessageInfo struct{}](https://github.com/golang/protobuf/blob/1b794fe86dd6a0c7c52ae69b5c9cb0aeedc52afa/proto/deprecated.go#L86) has been changed, all methods of this struct have been disabled by raising a panic.
```
func (*InternalMessageInfo) Size(Message) int { panic("not implemented") }
```

Let me try to explain why this panic is triggered by https://github.com/kubernetes/kubernetes/blob/863ce9726e0aa617f03332f42ff3cebb7abe8fc0/staging/src/k8s.io/client-go/discovery/discovery_client_test.go#L474
When the function `github.com/gogo/protobuf/proto.Marshal(*openapi_v2.Document)` marshal an openapi document, i will call [github.com/googleapis/gnostic/openapiv2.Document->XXX_Size()](https://github.com/googleapis/gnostic/blob/896953e6749863beec38e27029c804e88c3144b8/openapiv2/OpenAPIv2.pb.go#L550).

The thing is `XXX_Size()` depends on `golang/protobuf/proto.InternalMessageInfo`.
```
var xxx_messageInfo_Document proto.InternalMessageInfo
func (m *Document) XXX_Size() int {
	return xxx_messageInfo_Document.Size(m)
}
```
Even though we didn't change `github.com/gogo/protobuf` and `github.com/googleapis/gnostic`, but we can't use `golang/protobuf/proto.InternalMessageInfo` anymore as it has been deprecated(panic).

So, the third commits just simply changed the marshal function from `github.com/gogo/protobuf` to `github.com/golang/protobuf`.
  
@liggitt

RainbowMango replied:
> we use github.com/gogo/protobuf in many other places, are any of them affected?

There are only two places used gogo/protobuf/proto.Marshal() and I tried to fix them in the third commit.

I don't know if there are any potential risks. I didn't mean to remove gogo/protobuf in this PR, but after fixing the panic, it seems client-go doesn't need it anymore.

liggitt replied:
is the issue that we have stale generated code referencing proto.InternalMessageInfo?

There are many instances of that still in the codebase:

```
vendor/google.golang.org/grpc/health/grpc_health_v1/health.pb.go:	proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/grpc/health/grpc_health_v1/health.pb.go:var xxx_messageInfo_HealthCheckRequest proto.InternalMessageInfo
--
vendor/google.golang.org/grpc/binarylog/grpc_binarylog_v1/binarylog.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/grpc/binarylog/grpc_binarylog_v1/binarylog.pb.go:var xxx_messageInfo_GrpcLogEntry proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/app_identity/app_identity_service.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/app_identity/app_identity_service.pb.go:var xxx_messageInfo_AppIdentityServiceError proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/remote_api/remote_api.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/remote_api/remote_api.pb.go:var xxx_messageInfo_Request proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/datastore/datastore_v3.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/datastore/datastore_v3.pb.go:var xxx_messageInfo_Action proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/log/log_service.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/log/log_service.pb.go:var xxx_messageInfo_LogServiceError proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/modules/modules_service.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/modules/modules_service.pb.go:var xxx_messageInfo_ModulesServiceError proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/urlfetch/urlfetch_service.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/urlfetch/urlfetch_service.pb.go:var xxx_messageInfo_URLFetchServiceError proto.InternalMessageInfo
--
vendor/google.golang.org/appengine/internal/base/api_base.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/google.golang.org/appengine/internal/base/api_base.pb.go:var xxx_messageInfo_StringProto proto.InternalMessageInfo
--
vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client/client.pb.go:	proto "github.com/golang/protobuf/proto"
vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client/client.pb.go:var xxx_messageInfo_Packet proto.InternalMessageInfo
--
vendor/github.com/grpc-ecosystem/grpc-gateway/internal/stream_chunk.pb.go:import proto "github.com/golang/protobuf/proto"
vendor/github.com/grpc-ecosystem/grpc-gateway/internal/stream_chunk.pb.go:var xxx_messageInfo_StreamError proto.InternalMessageInfo
--
vendor/github.com/googleapis/gnostic/extensions/extension.pb.go:	proto "github.com/golang/protobuf/proto"
vendor/github.com/googleapis/gnostic/extensions/extension.pb.go:var xxx_messageInfo_Version proto.InternalMessageInfo
--
vendor/github.com/googleapis/gnostic/openapiv2/OpenAPIv2.pb.go:	proto "github.com/golang/protobuf/proto"
vendor/github.com/googleapis/gnostic/openapiv2/OpenAPIv2.pb.go:var xxx_messageInfo_AdditionalPropertiesItem proto.InternalMessageInfo
--
vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:	proto "github.com/golang/protobuf/proto"
vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:var xxx_messageInfo_GetPluginInfoRequest proto.InternalMessageInfo
```

are all of those potentially panic-inducing now?

RainbowMango replied:
> are all of those potentially panic-inducing now?

Yes! Awesome! I didn't check the generated code, and I'm glad you point it out.

Do you have any ideas about how to solve this?
IMO, given [github.com/golang/protobuf has been superseded by the google.golang.org/protobuf module](https://github.com/golang/protobuf#go-support-for-protocol-buffers), 
And, the recommendation from  github.com/golang/protobuf as follows:
> Versions v1.4 and later of github.com/golang/protobuf are implemented in terms of google.golang.org/protobuf. Programs which use both modules must use at least version v1.4 of this one.

I didn't check every module you list above, but I did found some of them using `protoc-gen-go`'s version lower than v1.4. such as:
https://github.com/googleapis/gnostic/blob/428ada5e5547af95eb370db4c0528778062dfbc1/openapiv2/OpenAPIv2.pb.go#L17-L21

I think, the `*.pb.go` in the vendor modules, should be regenerated by protoc-gen-go@v1.4.0+, or regernerated by the replacement [protogen](https://github.com/protocolbuffers/protobuf-go/tree/master/compiler/protogen). How do you say?

By the way, I still have a question in my mind.
I can't figure out why no panic found In this PR's CI jobs? Does that mean some checkpoint missing?

liggitt replied:
> I think, the `*.pb.go` in the vendor modules, should be regenerated by protoc-gen-go@v1.4.0+, or regernerated by the replacement [protogen](https://github.com/protocolbuffers/protobuf-go/tree/master/compiler/protogen). How do you say?

We do not modify the content of the `vendor` dir, so if changes are required in our dependencies, that must start as a PR to the upstream project(s), then an update in k8s.io/kubernetes once that is merged and released

> I can't figure out why no panic found In this PR's CI jobs? Does that mean some checkpoint missing?

We don't run all the unit tests of all our dependencies. I expect there would be more detailed coverage in those that could expose some of these issues.

dsnet replied:
See my comment here: https://github.com/golang/protobuf/issues/1126#issuecomment-628375275

If it helps, we can cut a `github.com/golang/protobuf@v1.4.2` release that adds back minimal support for `InternalMessageInfo`. The fundamental problem is that the `gogo/protobuf` implementation violates documented guidelines to **not** call certain API endpoints.

RainbowMango replied:
Awesome! It works with your patch, after I discard the third commit and run following commands, unit test goes all well.
```
# hack/pin-dependency.sh github.com/golang/protobuf f183159caab1d3a7dbf85306153a71aed61e9d8d
# hack/update-vendor.sh
# go test ./vendor/k8s.io/client-go/discovery/
ok  	k8s.io/client-go/discovery	0.033s
```

Thanks, @dsnet for your kindly support. 

So, if it acceptable for you to go with `github.com/golang/protobuf@v1.4.2`?

RainbowMango commented:
@liggitt Do you have any concerns?

RainbowMango commented:
/test pull-kubernetes-files-remake

RainbowMango commented:
No new changes, just rebase and solve conflicts. @liggitt

RainbowMango commented:
/test pull-kubernetes-integration

RainbowMango commented:
/test pull-kubernetes-integration

RainbowMango commented:
/test pull-kubernetes-e2e-kind-ipv6

RainbowMango commented:
/test pull-kubernetes-integration
/test pull-kubernetes-e2e-gce

RainbowMango commented:
/test pull-kubernetes-integration

RainbowMango commented:
@liggitt Cloud you please take a look again with `github.com/golang/protobuf@v1.4.2`?

liggitt commented:
this looks better, thanks. needs a rebase, then lgtm

RainbowMango commented:
/test pull-kubernetes-dependencies
/test pull-kubernetes-e2e-kind

It's a little weird that `hack/update-vendor.sh` looks good locally, and I run with Go v1.13.9 and v1.14.2.  Just retry this time.

```
--- vendor/k8s.io/sample-apiserver/go.sum	2020-05-18 01:32:26.510538144 +0000
+++ /home/prow/go/src/k8s.io/kubernetes/_tmp/kube-vendor.UYm3gQ/kubernetes/vendor/k8s.io/sample-apiserver/go.sum	2020-05-18 01:33:52.972131844 +0000
@@ -49,6 +49,7 @@ github.com/bgentry/speakeasy v0.1.0/go.m
 github.com/blang/semver v3.5.0+incompatible h1:CGxCgetQ64DKk7rdZ++Vfnb1+ogGNnB17OJKJXD2Cfs=
 github.com/blang/semver v3.5.0+incompatible/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
 github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
 github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/cespare/xxhash/v2 v2.1.1 h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=
 github.com/cespare/xxhash/v2 v2.1.1/go.mod h1:VGX0DQ3Q6kWi7AoAeZDth3/j3BFtOZR5XLFGgcrjCOs=
Vendor Verify failed.
If you're seeing this locally, run the below command to fix your directories:
hack/update-vendor.sh
Skipping golang.org/x/... dependencies, pass --all to include
All pinned versions of checked dependencies match their preferred version.
```

liggitt commented:
Make sure you're rebased on latest master?

RainbowMango commented:
> Make sure you're rebased on latest master?

I think so. But there're two more commits merged on master, I'll rebase again and give it a try.

```
# git log
...
commit 931080fc95daa5ce4c792e98d01488aa674eb049
Author: RainbowMango <renhongcai@huawei.com>
Date:   Mon May 18 09:25:43 2020 +0800

    bump github.com/prometheus/client_golang to v1.6.0

commit c6147e3231298b58ffb9c65e1a107768b2447de0
Merge: 55d7153 124386e
Author: Kubernetes Prow Robot <k8s-ci-robot@users.noreply.github.com>
Date:   Sun May 17 16:09:36 2020 -0700

    Merge pull request #91170 from prasadkatti/fix_ident
    
    Fix identation in kubeadm api docs
```

RainbowMango committed:
bump github.com/prometheus/client_golang to v1.6.0

RainbowMango committed:
Pin transitive dependencies:
pin dependency github.com/beorn7/perks from v1.0.0 to v1.0.1
pin dependency github.com/golang/protobuf from v1.3.3 to v1.4.2
pin denpendency github.com/json-iterator/go from v1.1.8 to v1.1.9
pin dependency github.com/prometheus/common from v0.4.1 to v0.9.1
pin dependency github.com/prometheus/procfs from v0.0.5 to v0.0.11
pin dependency github.com/alecthomas/template from v0.0.0-20160405071501-a0175ee3bccc to v0.0.0-20190718012654-fb15b899a751
pin dependency github.com/alecthomas/units from v0.0.0-20151022065526-2efee857e7cf to v0.0.0-20190717042225-c3de453c63f4
pin dependency github.com/go-kit/kit from v0.8.0 to v0.9.0
pin dependency github.com/go-logfmt/logfmt from v0.3.0 to v0.4.0

Co-Authored-By: Jordan Liggitt <jordan@liggitt.net>

RainbowMango commented:
Failed again... @liggitt  any idea about this?

liggitt commented:
when I check out your branch and run `hack/update-vendor.sh`, I get this:

```diff
diff --git a/staging/src/k8s.io/apiextensions-apiserver/go.sum b/staging/src/k8s.io/apiextensions-apiserver/go.sum
index 51ce932cd36..538a10f25a6 100644
--- a/staging/src/k8s.io/apiextensions-apiserver/go.sum
+++ b/staging/src/k8s.io/apiextensions-apiserver/go.sum
@@ -55,6 +55,7 @@ github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kB
 github.com/blang/semver v3.5.0+incompatible h1:CGxCgetQ64DKk7rdZ++Vfnb1+ogGNnB17OJKJXD2Cfs=
 github.com/blang/semver v3.5.0+incompatible/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
 github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
 github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/cespare/xxhash/v2 v2.1.1 h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=
 github.com/cespare/xxhash/v2 v2.1.1/go.mod h1:VGX0DQ3Q6kWi7AoAeZDth3/j3BFtOZR5XLFGgcrjCOs=
diff --git a/staging/src/k8s.io/kube-aggregator/go.sum b/staging/src/k8s.io/kube-aggregator/go.sum
index bd2349b28c9..3ab75d37b53 100644
--- a/staging/src/k8s.io/kube-aggregator/go.sum
+++ b/staging/src/k8s.io/kube-aggregator/go.sum
@@ -49,6 +49,7 @@ github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kB
 github.com/blang/semver v3.5.0+incompatible h1:CGxCgetQ64DKk7rdZ++Vfnb1+ogGNnB17OJKJXD2Cfs=
 github.com/blang/semver v3.5.0+incompatible/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
 github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
 github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/cespare/xxhash/v2 v2.1.1 h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=
 github.com/cespare/xxhash/v2 v2.1.1/go.mod h1:VGX0DQ3Q6kWi7AoAeZDth3/j3BFtOZR5XLFGgcrjCOs=
diff --git a/staging/src/k8s.io/sample-apiserver/go.sum b/staging/src/k8s.io/sample-apiserver/go.sum
index acaff528ecb..bd389514b35 100644
--- a/staging/src/k8s.io/sample-apiserver/go.sum
+++ b/staging/src/k8s.io/sample-apiserver/go.sum
@@ -49,6 +49,7 @@ github.com/bgentry/speakeasy v0.1.0/go.mod h1:+zsyZBPWlz7T6j88CTgSN5bM796AkVf0kB
 github.com/blang/semver v3.5.0+incompatible h1:CGxCgetQ64DKk7rdZ++Vfnb1+ogGNnB17OJKJXD2Cfs=
 github.com/blang/semver v3.5.0+incompatible/go.mod h1:kRBLl5iJ+tD4TcOOxsy/0fnwebNt5EWlYSAyrTnjyyk=
 github.com/census-instrumentation/opencensus-proto v0.2.1/go.mod h1:f6KPmirojxKA12rnyqOA5BBL4O983OfeGPqjHWSTneU=
+github.com/cespare/xxhash v1.1.0 h1:a6HrQnmkObjyL+Gs60czilIUGqrzKutQD6XZog3p+ko=
 github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/cespare/xxhash/v2 v2.1.1 h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=
 github.com/cespare/xxhash/v2 v2.1.1/go.mod h1:VGX0DQ3Q6kWi7AoAeZDth3/j3BFtOZR5XLFGgcrjCOs=
```

I pushed a squashed commit containing your two commits and that fix to https://github.com/liggitt/kubernetes/commits/pr_bump_client_golang_1_6

RainbowMango commented:
Thanks @liggitt , I appreciate your help. I removed all cached modules and re-run `hack/update-vendor.sh`, seems it's fine now. And I added you as the co-author too.

liggitt commented:
/lgtm
/approve

beorn7 commented:
Wow, that was quite a ride. Thanks, everyone, for the effort.
